presto-docs/src/main/sphinx/release/release-341.md

```diff
@@ -0,0 +1,164 @@
+* Fix serialization of `NULL` values in `ROW`, `MAP` and `ARRAY` types for old Presto clients. ({issue}`4778`)
+* Fix failure when aggregation query contains duplicate expressions. ({issue}`4872`)
+* Fix compiler failure when querying timestamps with a precision greater than 6. ({issue}`4824`)
+* Change timestamp operations to match the SQL specification. The value of a `TIMESTAMP` type is not affected by the session
+  time zone. ({issue}`4799`)
+* Fix parsing failure of timestamps due to daylight saving changes. ({issue}`4799`)
+* Remove the `deprecated.legacy-timestamp` configuration property and the `legacy_timestamp` session property. ({issue}`4799`)
+* Preserve precision when applying `AT TIME ZONE` to values of type `TIMESTAMP`. ({issue}`4866`)
+* Fix failure when calling `EXTRACT` with `TIMEZONE_HOUR` and `TIMEZONE_MINUTE` for `TIMESTAMP WITH TIME ZONE` type. ({issue}`4867`)
+* Fix query deadlock for connectors that wait for dynamic filters. ({issue}`4946`)
+* Add [](optimizer_rule_stats) system table. ({issue}`4659`)
+* Fix failure when `TIME` or `TIMESTAMP` subtraction returns a negative value. ({issue}`4847`)
+* Fix failure when duplicate expressions appear in `DISTINCT` clause. ({issue}`4787`)
+* Fix failure for join queries during spilling or when available memory is low. ({issue}`4994`)
+* Fix issue where ``query_max_scan_physical_bytes`` session property was ignored if
+  ``query.max-scan-physical-bytes`` config property was not defined. ({issue}`5009`)
+* Fix incorrect sampling ratio when `TABLESAMPLE` is used with percentage fraction. ({issue}`5074`)
+* Correct rejecting queries using `TABLESAMPLE` with ratio being not a number. ({issue}`5074`)
+* Fail with an explicit error rather than `OutOfMemoryError` for certain operations. ({issue}`4890`)
+
+## Security Changes
+
+* Add [Salesforce password authentication](/security/salesforce). ({issue}`4372`)
+* Add support for interpolating [secrets](/security/secrets) into access-control.properties. ({issue}`4854`)
+* Only request HTTPS client certificate when certificate authentication is enabled. ({issue}`4804`)
+* Add user mapping configuration to preserve lower case or upper case user names. ({issue}`4736`)
+
+## Web UI
+
+* Fix displaying physical input read time in detailed query view. ({issue}`4962`)
+
+## JDBC Driver Changes
+
+* Implement `ResultSet.getStatement()`. ({issue}`4957`)
+
+## BigQuery Connector Changes
+
+* Add support for HOURLY partitioning of tables. ({issue}`4968`)
+* Redact value of `bigquery.credentials-key` from logs. ({issue}`4968`)
+
+## Cassandra Connector Changes
+
+* Map Cassandra `TIMESTAMP` type to Presto `TIMESTAMP(3) WITH TIME ZONE` type. ({issue}`2269`)
+
+## Hive Connector Changes
+
+* Skip stripes and row group based on timestamp statistics for ORC files. ({issue}`1147`)
+* Skip DeepArchive objects when `skip-glacier-objects` is set. ({issue}`5002`)
+* Use temporary staging directory for write operation to sorted bucketed table. ({issue}`3434`)
```

Comment:
Can we say what's the user visible impact of the change, in case user does not fully appreciate what the staging directory is?

Reply:
@kokosing

Reply:
>  in case user does not fully appreciate what the staging directory is?

Maybe we should explain what staging is in the docs first? Then we could reference that here. Do we need that? If user does not care (or know) about staging then I guess they might just ignore it.